edgelet/iotedged/src/lib.rs

```diff
@@ -264,49 +280,80 @@ impl Main {
                     let code = start_api(
```

Comment:
Since you're making this common for all three types, please also fix this abomination of a loop to:

```rust
loop {
    let code = ...
    if code != StartApiReturnStatus::Restart {
        break;
    }
}
```